audio_service/lib/audio_service.dart

```diff
@@ -523,11 +548,15 @@ class Rating {
   String toString() => '${_toMessage().toMap()}';
 
   @override
-  int get hashCode => toString().hashCode;
+  int get hashCode => hashValues(_value, _type);
 
   @override
-  bool operator ==(dynamic other) =>
-      other is Rating && _type == other._type && _value == other._value;
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType) {
```

Comment:
Hmm, I know the Flutter team likes to do that, although I have no idea why they like to use procedural code for something that is easily a function.

Reply:
Can elaborate on that?

Reply:
I mean I prefer the arrow notation for functions:

```dart
  bool operator ==(Object other) =>
      other.runtimeType == runtimeType &&
      other is Rating &&
      _type == other._type &&
      _value == other._value;
```

Reply:
feel free to change that in the way you want, I don't have any objections